report-viewer/src/components/ComparisonsTable.vue

```diff
@@ -2,121 +2,233 @@
+          isAnonymous(item.firstSubmissionId)
             ? "Hidden"
-            : displayName(comparison.firstSubmissionId)
+            : displayName(item.firstSubmissionId)
         }}
       </td>
       <td
         @click="
           navigateToComparisonView(
-            comparison.firstSubmissionId,
-            comparison.secondSubmissionId
+            item.firstSubmissionId,
+            item.secondSubmissionId
           )
         "
+        class="td3"
       >
         <img alt=">>" src="@/assets/double_arrow_black_18dp.svg" />
       </td>
       <td
         @click="
           navigateToComparisonView(
-            comparison.firstSubmissionId,
-            comparison.secondSubmissionId
+            item.firstSubmissionId,
+            item.secondSubmissionId
           )
         "
         :class="{
-          'anonymous-style': isAnonymous(comparison.secondSubmissionId),
+          'anonymous-style': isAnonymous(item.secondSubmissionId),
         }"
+        class="td4"
       >
         {{
-          isAnonymous(comparison.secondSubmissionId)
+          isAnonymous(item.secondSubmissionId)
             ? "Hidden"
-            : displayName(comparison.secondSubmissionId)
+            : displayName(item.secondSubmissionId)
         }}
       </td>
       <td
         @click="
           navigateToComparisonView(
-            comparison.firstSubmissionId,
-            comparison.secondSubmissionId
+            item.firstSubmissionId,
+            item.secondSubmissionId
           )
         "
+        class="td5"
       >
-        {{ formattedMatchPercentage(comparison.similarity) }}
+        {{ formattedMatchPercentage(item.similarity) }}
       </td>
-      <td>
+      <td style="width: 5%">
         <img
           v-if="
             isInCluster(
-              comparison.firstSubmissionId,
-              comparison.secondSubmissionId
+              item.firstSubmissionId,
+              item.secondSubmissionId
             )
           "
           alt=">>"
           src="@/assets/keyboard_double_arrow_down_black_18dp.svg"
-          @click="toggleDialog(index)"
+          @click="toggleDialog(item.id-1)"
         />
       </td>
       <GDialog
         v-if="
           isInCluster(
-            comparison.firstSubmissionId,
-            comparison.secondSubmissionId
+            item.firstSubmissionId,
+            item.secondSubmissionId
           )
         "
-        v-model="dialog[index]"
+        v-model="dialog[item.id-1]"
       >
         <ClustersList
           :clusters="
             getClustersFor(
-              comparison.firstSubmissionId,
-              comparison.secondSubmissionId
+              item.firstSubmissionId,
+              item.secondSubmissionId
             )
           "
-          :comparison="comparison"
+          :comparison="item"
         />
       </GDialog>
-    </tr>
-  </table>
+      </tr>
+      <tr v-else class="selectableOdd">
```

Comment:
Having the same code twice only because of the highlighting is code duplication. Please use something like the deleted `:nth-child` instead for highlighting and do not duplicate the code.

Reply:
`:nth-child` works not for this situation. Each comparison is a table, so each comparison only has one `<tr>` element.